project/HmrcBuild.scala

```diff
@@ -47,10 +48,16 @@ object HmrcBuild extends Build {
         Resolver.bintrayRepo("hmrc", "releases"),
         "typesafe-releases" at "http://repo.typesafe.com/typesafe/releases/"
       ),
-      crossScalaVersions := Seq("2.11.7")
+      crossScalaVersions := Seq("2.11.7"),
+      testGrouping in sbt.Test := oneForkedJvmPerTest((definedTests in sbt.Test).value)
     )
     .settings(TwirlKeys.templateImports ++= Seq("play.api.mvc._", "play.api.data._", "play.api.i18n._", "play.api.templates.PlayMagic._"))
     .settings(unmanagedSourceDirectories in sbt.Compile += baseDirectory.value / "src/main/twirl")
+
+  private def oneForkedJvmPerTest(tests: Seq[TestDefinition]) =
```

Comment:
@feedmypixel - I could see two problems in your suggested approach - 

1. You have advised to write tests against `optimizelySnippet.scala.html`. So, if we then comment out the line where we render`@optimizelySnippet()` in `head.scala.html`, no test will fail because we will be having tests only against `optimizelySnippet.scala.html`. In my approach, I have actually tested the `head.scala.html` in my original PR which is the right thing to test, after all we want head.scala.html to behave as expected.

2. Second problem is more around design. You have advised to introduce a parameter `idOpt: Option[String]` in `optimizelySnippet.scala.html` which will be only be exercised in tests. In actual runtime, when this template will be rendered in `head.scala.html`, the parameter will not be passed. This means we have added code to production code which will be used only in tests. This sounds like a code smell to me.

If you still want to go ahead with your suggested approach with the problems which I have highlighted above, then let me know, I'll make the changes as you have advised. After all, it is your team who has to maintain this codebase.